### DISCUSSION TITLE
How to integrate bootstrap 5 with nuxt 3?

### DISCUSSION THREAD

**Discussion by shereali** [nuxt/nuxt#16041] [Questions]
Hi, I have read some articles and watched videos on bootstrap 5 integrated with nuxt 3.  But I did not get any good solution on it. I have tried many ways but bootstrap.min.js is not working. Help me, please.

**Comment by Goldenback** [comment]
Hi, i tried too but i can't get it to work properly without breaking everything.
Did you find a way ?

**Comment by valcosmos** [comment]
This repo might help you.

[https://github.com/Debonex/samples/tree/master/nuxt3-bootstrap5](https://github.com/Debonex/samples/tree/master/nuxt3-bootstrap5)

**Reply by Goldenback** [reply]
Thanks i did my own Boiler since this one is a bit buggy to use.
https://github.com/GoriIIe/Nuxt3-Boilerplate-Digityser

**Comment by basicactor** [comment]
Hi, if you have not solved yet, try this way. It works for me. 

1. Install bootstrap5 
```
yarn add bootstrap
```

2. Change the nuxt config settings
```ts:nuxt.config.ts
import { defineNuxtConfig } from "nuxt";

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
  css: ["bootstrap/dist/css/bootstrap.min.css"], // add
  vite: {
    define: {
      "process.env.DEBUG": false,
    },
  },
});

```

3. See if the styles is applied.
```vue:app.vue
<template>
  <div class="m-5">
    <!-- <NuxtPage /> -->
    <div class="btn btn-success">Success</div>
    <div class="btn btn-danger">Danger</div>
    <div class="btn btn-warning">Warning</div>
  </div>
</template>
```

**Reply by hafizjavaid** [reply]
It's working for css but not for js files

**Reply by lacorde** [reply]
@hafizjavaid  for boostrap js, the only way I found is adding the cdn in your nuxt.config, as follow :
```
app: {
    head: {
      script: [
        {
          src: "https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/js/bootstrap.bundle.min.js",
          type: "text/javascript",
        },
      ],
    },
  },
```
This will not work with SSR atm, as support has been forwarded to Bootstrap 5.3.

**Comment by Chuiantw1212** [comment]
I had similar questions.
It turns out I did not assign .client prefix to /plugins/bootstrap.js.
After I change the file name from bootstrap.js to boostrap.client.js,
It is working.

**Reply by dkuye** [reply]
Hi, 
what do you have inside boostrap.client.js file?

**Comment by kmukku** [comment]
You can also use bootstrap as a plugin. See https://v3.nuxtjs.org/guide/directory-structure/plugins/#typing-plugins

I have a plugin called bootstrap.client.ts with following:

```
import * as bootstrap from 'bootstrap'

export default defineNuxtPlugin(() => {
    return {
      provide: {
        bootstrap: bootstrap
      }
    }
})
```

and when I need it I use 

`const { $bootstrap } = useNuxtApp();`

and

`const myModal = new $bootstrap.Modal(document.getElementById('exampleModal'));`

**Reply by sizoroot** [reply]
This works for me! Ty!

**Reply by ak-umka** [reply]
for every page i have to import bootstrap ?

**Reply by DykiSA** [reply]
> for every page i have to import bootstrap ?

No, it is only if you need javascript function, e.g. modal

**Comment by wobsoriano** [comment]
https://github.com/cdmoro/bootstrap-vue-3
